cmake_modules/ParsecCompilerFlags.cmake

```diff
@@ -187,9 +186,9 @@ endif()
 if(CMAKE_Fortran_COMPILER_WORKS)
   get_filename_component (Fortran_COMPILER_NAME ${CMAKE_Fortran_COMPILER} NAME)
   #  message(STATUS "Fortran Compiler ${Fortran_COMPILER_NAME} id is ${CMAKE_Fortran_COMPILER_ID}")
-  if(${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU")
+  if(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
     # gfortran or g77
-    if(${Fortran_COMPILER_NAME} MATCHES g77)
+    if("${Fortran_COMPILER_NAME}" MATCHES g77)
```

Comment:
Looks to me as if `MATCHES` works with variable names too: https://cmake.org/cmake/help/latest/command/if.html#variable-expansion